torch/distributed/fsdp/_init_utils.py

```diff
@@ -98,21 +99,36 @@ def _init_process_group_state(
     process_group: ProcessGroupType,
     sharding_strategy: ShardingStrategy,
     policy: Optional[_Policy],
+    device_mesh: Optional[DeviceMesh] = None,
 ) -> _FSDPState:
+    if process_group is not None and device_mesh is not None:
+        raise ValueError(
+            "Cannot pass both process_group and device_mesh at the "
+            "same time. Please just pass only one of them."
+        )
     is_hybrid_strategy = sharding_strategy in HYBRID_SHARDING_STRATEGIES
     if is_hybrid_strategy:
-        if process_group is None and policy is None:
+        if process_group is None and policy is None and device_mesh is None:
             # Raise an error here, since this is manual wrapping with no process group
             # passed in, there is no way to ensure all wrapped FSDP instances use the same
             # process groups.
             raise ValueError(
-                f"Manual wrapping with {sharding_strategy} requires explicit specification of process group."
+                f"Manual wrapping with {sharding_strategy}",
+                "requires explicit specification of process group or device_mesh.",
+            )
+        else:
+            state = _init_process_group_state_for_hybrid_shard(
+                state, process_group, device_mesh
             )
-        state = _init_process_group_state_for_hybrid_shard(state, process_group)
     else:
-        state.process_group = (
-            process_group if process_group is not None else _get_default_group()
-        )
+        if device_mesh:
+            state._device_mesh = device_mesh
+            state.process_group = device_mesh.get_dim_groups(mesh_dim=0)
```

Comment:
Is the `mesh_dim` always 0 or this is just a temporarily case for 1D? If it is for later, we may want to postpone the implementation, this `if`, until the 2D PRs. This is not going to affect HSDP which is the main purpose of this PR.

Reply:
HSDP would get taken care in _init_process_group_state_for_hybrid_shard() so this is only for FSDP 1D, so the mesh_dim would always be 0. 

For 2D, since we are planning to just be passing the device mesh for each parallelism, this would also hold true.

Reply:
uh you are right. Please ignore the comment.